geoindicators/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/geoindicators/SpatialUnits.groovy

```diff
@@ -351,53 +351,85 @@ IProcess createBlocks() {
+                datasource: JdbcDataSource
+        outputs outputTableName: String, idColumnTab2: String
+        run { inputTable1, inputTable2, idColumnTab1, idColumnTab2, prefixName, pointOnSurface,
+              nbRelations, datasource ->
 
             def GEOMETRIC_COLUMN_LOW = "the_geom"
             def GEOMETRIC_COLUMN_UP = "the_geom"
 
-            info "Creating the Tables of relations between the twos scales :  $inputLowerScaleTableName and $inputUpperScaleTableName"
+            info "Creating a spatial join between objects from two tables :  $inputTable1 and $inputTable2"
 
             // The name of the outputTableName is constructed (the prefix name is not added since it is already contained
             // in the inputLowerScaleTableName object
-            def outputTableName = postfix inputLowerScaleTableName, "corr"
-            datasource."$inputLowerScaleTableName".the_geom.createSpatialIndex()
-            datasource."$inputUpperScaleTableName".the_geom.createSpatialIndex()
+            def outputTableName = postfix "${inputTable1}_${inputTable2}", "join"
+            datasource."$inputTable1".the_geom.createIndex()
+            datasource."$inputTable2".the_geom.createIndex()
+            datasource."$inputTable2"[idColumnTab2].createIndex()
+
+            def query = """DROP TABLE IF EXISTS $outputTableName;"""
+            def subqueryNbRelations = ""
+            if (pointOnSurface){
+                query += """CREATE TABLE $outputTableName AS SELECT a.*, b.$idColumnTab2 
+                                    FROM $inputTable1 a, $inputTable2 b 
+                                    WHERE   ST_POINTONSURFACE(a.$GEOMETRIC_COLUMN_LOW) && st_force2d(b.$GEOMETRIC_COLUMN_UP) AND 
+                                            ST_INTERSECTS(ST_POINTONSURFACE(a.$GEOMETRIC_COLUMN_LOW), st_force2d(b.$GEOMETRIC_COLUMN_UP))"""
+            }
+            else {
+                def temporaryTab = "temporaryTab"
+                if(nbRelations!=null){
+                    subqueryNbRelations += """DESC LIMIT $nbRelations"""
+                }
+                query += """DROP TABLE IF EXISTS $temporaryTab;
+                            CREATE TABLE $temporaryTab 
+                                    AS SELECT   a.*, b.$idColumnTab2,
+                                                ST_AREA(ST_INTERSECTION(st_force2d(st_makevalid(a.$GEOMETRIC_COLUMN_LOW)), 
+                                                st_force2d(st_makevalid(b.$GEOMETRIC_COLUMN_UP)))) AS AREA
+                                    FROM    $inputTable1 a, $inputTable2 b
+                                    WHERE   a.$GEOMETRIC_COLUMN_LOW && b.$GEOMETRIC_COLUMN_UP AND 
+                                            ST_INTERSECTS(st_force2d(a.$GEOMETRIC_COLUMN_LOW), st_force2d(b.$GEOMETRIC_COLUMN_UP));
+                            CREATE INDEX IF NOT EXISTS id_area ON $temporaryTab USING BTREE(AREA);
+                            CREATE TABLE $outputTableName 
```

Comment:
Use the same approach as  :   

```
 datasource """DROP TABLE IF EXISTS $outputTableName;
                     CREATE TABLE $outputTableName AS SELECT a.*, (SELECT b.$idColumnUp 
                     FROM $inputUpperScaleTableName b WHERE a.$GEOMETRIC_COLUMN_LOW && b.$GEOMETRIC_COLUMN_UP AND 
                     ST_INTERSECTS(st_force2d(a.$GEOMETRIC_COLUMN_LOW), st_force2d(b.$GEOMETRIC_COLUMN_UP)) ORDER BY 
                     ST_AREA(ST_INTERSECTION(st_force2d(st_makevalid(a.$GEOMETRIC_COLUMN_LOW)), st_force2d(st_makevalid(b.$GEOMETRIC_COLUMN_UP)))) 
                     DESC LIMIT 1) AS $idColumnUp FROM $inputLowerScaleTableName a """
```

Reply:
I have tried adding to the query:
```sql
(SELECT ST_AREA(ST_INTERSECTION(st_force2d(st_makevalid(a.$GEOMETRIC_COLUMN_LOW)), st_force2d(st_makevalid(b.$GEOMETRIC_COLUMN_UP)))) AS AREA 
                     FROM $inputUpperScaleTableName b WHERE a.$GEOMETRIC_COLUMN_LOW && b.$GEOMETRIC_COLUMN_UP AND 
                     ST_INTERSECTS(st_force2d(a.$GEOMETRIC_COLUMN_LOW), st_force2d(b.$GEOMETRIC_COLUMN_UP)) ORDER BY 
                     ST_AREA(ST_INTERSECTION(st_force2d(st_makevalid(a.$GEOMETRIC_COLUMN_LOW)), st_force2d(st_makevalid(b.$GEOMETRIC_COLUMN_UP)))) 
                     DESC LIMIT 1) AS $idColumnUp```

but I got a "multiple row issue"